woocommerce/woocommerce-ios
# Widgets: Keychain Setup

Part of #7564 

# Why

In order to fetch data in the widget extension from the network, we need to have access to the auth token which should be shared in the keychain for security.

# How 

- Adds keychain entitelment.
- Set the authToken to the keychain when the app starts, logins, or switches store.
- Render a temporary "Authenticated" label in the widget if the widget extension can reach the access token.

# Screenshot

Logged in | Not logged in
---- | ----
<img width="365" alt="loged-in" src="https://user-images.githubusercontent.com/562080/188980475-1eb4918f-f719-417b-9280-1e4d4d99146a.jpeg">|<img width="365" alt="not-loged-in" src="https://user-images.githubusercontent.com/562080/188980369-313b5f93-6abe-4372-96f8-dcf14fef45d1.png">


# Testing Steps

- Launch the app.
- Launch the widget extension
- See that the widget show the authenticated / not authenticated label
